File: emoji-library/src/main/java/br/com/instachat/emojilibrary/model/layout/EmojiCompatActivity.java
package br.com.instachat.emojilibrary.model.layout;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

/**
 * Created by edgar on 18/02/2016.
 */
public class EmojiCompatActivity extends AppCompatActivity {

    private OnBackPressedListener mOnBackPressedListener;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onBackPressed() {
        if (this.mOnBackPressedListener != null) {
            if (!this.mOnBackPressedListener.onBackPressed()) {
                super.onBackPressed();
            }
        } else {
            super.onBackPressed();
        }
    }

    public void setOnBackPressed(OnBackPressedListener backListener) {
        this.mOnBackPressedListener = backListener;
    }

    public interface OnBackPressedListener {
        Boolean onBackPressed();
    }
}

File: emoji-library/src/main/kotlin/br/com/instachat/emojilibrary/view/EmojiCompatActivity.kt
package br.com.instachat.emojilibrary.view

import androidx.appcompat.app.AppCompatActivity


open class EmojiCompatActivity : AppCompatActivity() {

    private var onBackPressedListener: OnBackPressedListener? = null

    override fun onBackPressed() {
        if (onBackPressedListener != null) {
            if (!onBackPressedListener!!.onBackPressed()!!) {
                super.onBackPressed()
            }
        } else {
            super.onBackPressed()
        }
    }

    open fun setOnBackPressed(backListener: OnBackPressedListener) {
        this.onBackPressedListener = backListener
    }


    interface OnBackPressedListener {
        fun onBackPressed(): Boolean?
    }
}
